tonic_validate/services/litellm_service.py

```diff
@@ -0,0 +1,140 @@
+        """
+        try:
+            self.check_environment(model)
+        except Exception as e:
+            logger.error(f"Error: {str(e)}")
+            raise e
+        self.model = model
+        self.encoder = encoder
+        self.max_retries = max_retries
+        self.exp_delay_base = exp_delay_base
+        self.starting_wait_time = starting_wait_time
+        self.cache = LLMCache()
+
+    def check_environment(self, model: str) -> None:
+        if "gemini" in model:
+            if "GEMINI_API_KEY" not in os.environ:
+                raise Exception(
+                    "GEMINI_API_KEY must be set in the environment when using Gemini"
+                )
+        if "claude" in model:
+            if "ANTHROPIC_API_KEY" not in os.environ:
+                raise Exception(
+                    "ANTHROPIC_API_KEY must be set in the environment when using Claude"
+                )
+        raise Exception("Model not supported. Please check the model name.")
+
+    async def get_response(self, prompt: str) -> str:
+        """
+        Retrieves a response from the language model asynchronously.
+
+        Parameters
+        ----------
+        prompt: str
+            The prompt to get a response for.
+
+        Returns
+        -------
+        str
+            The response from the language model.
+        """
+
+        async def get_litellm_response():
+            num_retries = 0
+            wait_time = self.starting_wait_time
+            while num_retries < self.max_retries:
+                try:
+                    messages = [
+                        {
+                            "role": "system",
+                            "content": "You are a helpful assistant. Respond using markdown.",
+                        },
+                        {"role": "user", "content": prompt},
+                    ]
+                    response = await acompletion(
+                        model=self.model,
+                        messages=messages,
+                        temperature=0.0,
+                    )
+                    # Check that type is ModelResponse
+                    if not isinstance(response, ModelResponse):
+                        raise Exception(
+                            f"Failed to get response from {self.model}, response is not a ModelResponse"
+                        )
+                    choice = response.choices[0]
+                    if not isinstance(choice, Choices):
+                        raise Exception(
+                            f"Failed to get response from {self.model}, choice is not a Choices object"
+                        )
+                    response_content = choice.message.content
+                    if response_content is None:
+                        raise Exception(
+                            "Failed to get a response, message does not exist"
+                        )
+                    return response_content
+                except Exception as e:
```

Comment:
FYI, litellm maps all exceptions from all providers to the openai exceptions. So we can reuse the logic there